src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -788,6 +788,10 @@ private static Expression CreateArgument(ParameterInfo parameter, RequestDelegat
         }
         else if (parameterCustomAttributes.OfType<FromKeyedServicesAttribute>().FirstOrDefault() is { } keyedServicesAttribute)
         {
+            if (factoryContext.ServiceProviderIsService is not IServiceProviderIsKeyedService)
+            {
+                throw new InvalidOperationException($"This service provider doesn't support keyed services.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Unable to resolve {nameof(FromKeyedServicesAttribute)} because the service provider doesn't support keyed services.");
```